File: src/main/java/me/fru1t/javafx/TextInputDialogUtils.java
package me.fru1t.javafx;

import javafx.scene.control.TextInputDialog;

import javax.annotation.Nullable;
import java.util.Optional;

/**
 * Utility functions for {@link TextInputDialog}
 */
public class TextInputDialogUtils {
    @Nullable
    public static String createShowAndWait(@Nullable String title, @Nullable String headerText,
            @Nullable String contentText) {
        TextInputDialog dialog = new TextInputDialog();
        dialog.setTitle(title);
        dialog.setHeaderText(headerText);
        dialog.setContentText(contentText);
        Optional<String> result = dialog.showAndWait();
        return result.orElse(null);
    }
}

File: src/main/kotlin/me/fru1t/javafx/TextInputDialogUtils.kt
package me.fru1t.javafx

import javafx.scene.control.TextInputDialog

/** Utility functions for [TextInputDialog]. */
object TextInputDialogUtils {
  fun createShowAndWait(title: String?, headerText: String?, contentText: String?): String? {
    val dialog = TextInputDialog()
    dialog.title = title
    dialog.headerText = headerText
    dialog.contentText = contentText
    val result = dialog.showAndWait()
    return result.orElse(null)
  }
}
